src/PixiEditor/Models/DocumentModels/UpdateableChangeExecutors/MagicWandToolExecutor.cs

```diff
@@ -25,27 +25,45 @@ public override ExecutionState Start()
         if (magicWand is null || members.Count == 0)
             return ExecutionState.Error;
 
-        mode = magicWand.SelectMode;
+        mode = magicWand.ResultingSelectionMode;
         memberGuids = members;
         considerAllLayers = magicWand.DocumentScope == DocumentScope.Canvas;
         if (considerAllLayers)
             memberGuids = document!.StructureHelper.GetAllLayers().Select(x => x.Id).ToList();
         var pos = controller!.LastPixelPosition;
         tolerance = (float)magicWand.Tolerance;
 
-        internals!.ActionAccumulator.AddActions(new MagicWand_Action(memberGuids, pos, mode, tolerance, document!.AnimationHandler.ActiveFrameBindable));
+        AddUpdateAction(pos);
 
         return ExecutionState.Success;
     }
 
+    public override void OnPixelPositionChange(VecI pos)
+    {
+        AddUpdateAction(pos);
+        internals!.ActionAccumulator.AddActions(new ChangeBoundary_Action());
```

Comment:
Do not use `ChangeBoundary_Action`, use `AddFinishedActions` instead. Although I don't think it should be finished here. One user drag should be considered one undo change in that case imo. 

You've correctly made MagicWand an `UpdateableChange`, so multiple changes are part of one change. If you change boundary on each movement, you remove the benefit of using `UpdateableChange`

Reply:
I am not sure why, I'd have to debug. Change_Boundary tells the undo system that undo packet should be closed. A packet can contain multiple changes and undo/redo them together. 

Some updateable changes also have logic that merges multiple changes into one.